src/internal/m365/collection/drive/collections.go

```diff
@@ -72,6 +72,70 @@ func NewCollections(
 	}
 }
 
+func deserializeAndValidateMetadata(
+	ctx context.Context,
+	cols []data.RestoreCollection,
+) (map[string]string, map[string]map[string]string, bool, error) {
+	deltas, prevs, canUse, err := DeserializeMetadata(ctx, cols)
+	if err != nil || !canUse {
+		return deltas, prevs, canUse, clues.Stack(err).OrNil()
+	}
+
+	// Go through and remove delta tokens if we didn't have any paths for them
+	// or one or more paths are empty (incorrect somehow). This will ensure we
+	// don't accidentally try to pull in delta results when we should have
+	// enumerated everything instead.
+	//
+	// Loop over the set of previous deltas because it's alright to have paths
+	// without a delta but not to have a delta without paths. This way ensures
+	// we check at least all the path sets for the deltas we have.
+	for drive := range deltas {
+		paths := prevs[drive]
+		if len(paths) == 0 {
+			delete(deltas, drive)
+		}
+
+		// Drives have only a single delta token. If we find any folder that
+		// seems like the path is bad we need to drop the entire token and start
+		// fresh. Since we know the token will be gone we can also stop checking
+		// for other possibly incorrect folder paths.
+		for _, prevPath := range paths {
+			if len(prevPath) == 0 {
+				delete(deltas, drive)
+				break
+			}
+		}
+	}
+
+	prevPathCollisions := map[string]string{}
+
+	for driveID, folders := range prevs {
```

Comment:
Could you please rearrange this? I find it confusing to initialize/clear the map this way instead of just initializing it in the first for-loop
```suggestion
	for driveID, folders := range prevs {
		prevPathCollisions := map[string]string{}

```

Reply:
Heh, yeah, I already did that in the other PR.   Had a "why did I even do this?" moment.